src/ConnectedMode/Migration/Interfaces.cs

```diff
@@ -33,7 +34,7 @@ internal interface IConnectedModeMigration
     {
         // TODO: decide if the method should return a list of specific manual cleanup
         // steps that the user still needs to perform
-        Task MigrateAsync(IProgress<MigrationProgress> progress, CancellationToken token);
+        Task MigrateAsync(BoundSonarQubeProject oldBinding, IProgress<MigrationProgress> progress, CancellationToken token);
```

Comment:
This is the class that actually does the work, so it is the thing that really needs the binding information.